x/mongo/driver/topology/topology.go

```diff
@@ -295,16 +307,38 @@ func (t *Topology) SelectServer(ctx context.Context, ss description.ServerSelect
 		defer ssTimeout.Stop()
 	}
 
-	sub, err := t.Subscribe()
-	if err != nil {
-		return nil, err
-	}
-	defer t.Unsubscribe(sub)
-
+	var doneOnce bool
+	var sub *driver.Subscription
+	selectionCfg := newServerSelectionConfig(ss, ssTimeoutCh)
 	for {
-		suitable, err := t.selectServer(ctx, sub.Updates, ss, ssTimeoutCh)
-		if err != nil {
-			return nil, err
+		var suitable []description.Server
+		var selectErr error
+
+		if !doneOnce {
+			// for the first pass, select a server from the current description
```

Comment:
```suggestion
                     // for the first pass, select a server from the current description.
		       // this improves connection speed with up-to-date topology descriptions
```

Reply:
I think the space might be wrong on my suggestion. Keep the spacing consistent :)

Reply:
done.